### ISSUE TITLE
"SC2054: Use spaces, not commas, to separate array elements" false positive

### ISSUE THREAD

**Issue by knweiss** [koalaman/shellcheck#91]
I ran into this case while shellchecking 3rd-party code:

``` bash
$ cat array.sh
#!/bin/bash
array=("one," "two,")
echo ${array[0]}
echo ${array[1]}
```

```
$ ./array.sh
one,
two,
```

```
$ shellcheck array.sh
In - line 2:
array=("one," "two,")
      ^-- SC2054: Use spaces, not commas, to separate array elements.
```

Isn't this valid syntax?

**Comment by ewancoder** [comment]
Should it warn on ```option=( rw,relatime,discard rw,relatime,discard rw,relatime rw,relatime,discard )``` ?

**Comment by deryni** [comment]
I'd argue that that's ambiguous (and thus deserves the warning).

It isn't necessarily clear which interpretation of that the user intended. Either space-parsing or comma-parsing might have been intended.

*You* know that that was intended to be

    declare -a option='([0]="rw,relatime,discard" [1]="rw,relatime,discard" [2]="rw,relatime" [3]="rw,relatime,discard")'

because you know what the options mean but someone could have written it intending for it to be

    declare -a option='([0]="rw" [1]="relatime" [2]="discard rw" [3]="relatime" [4]="discard rw" [5]="relatime rw" [6]="relatime" [7]="discard")'

instead. (Imagine if the line was `option=( user1,acl1,User Name 1,user2,acl2,User Name 2 )` for example.)
